docs/templates/contributing_template.md

```diff
@@ -0,0 +1,281 @@
+-->
+
+### Deliverables
+ 
+<!-- [PART7] Instructions:
+   * Kinds of tests - unit, integration & perf, a feature should accompany
+   * External API documentation updates
+   * Provide a snapshot of license header or a link to license header for files
+   * Commit message format or its specification
+
+MIOpen Example:
+1. Here are the key components required in kernel's pull request.
++ Each kernel needs to support these datatypes
+     - Until ROCm5.6,  fp16, bfloat16, fp32
+     - Starting ROCm5.7, f8 
++ Introduce the host-side API, if required
+E.g. For LayerNorm, https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/include/miopen/miopen.h
++ Add unit tests in gtest format
+Dir: https://github.com/ROCmSoftwarePlatform/MIOpen/tree/5c8118ee4339c2aae55f20241440af207e59d2a7/test/gtest
++ Include the relevant code in driver to measure per-kernel performance
+https://github.com/ROCmSoftwarePlatform/MIOpen/tree/master/driver
++ Post the performance # against the baseline kernel.
++ For new kernels in MIOpen, e.g. LayerNorm 
+   - obtain the baseline from ROCm/Pytorch build: https://github.com/ROCmSoftwarePlatform/pytorch/blob/main/aten/src
+/ATen/native/cuda/layer_norm_kernel.cu
+   - obtain the baseline from ROCm/Composable kernel library.
++ For overlapping kernels in MIOpen, 
+   - obtain the baseline by running MIOpen/driver code 
+
+1. For each new file in repository, 
+Please include the licensing header
+```
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 20xx Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+```
+-->
+
+### Process ###
+
+<!-- [PART8] Instructions:
+   * Outline the PR review process, most importantly, ensure list of reviewers are updated in Codeowners file 
+   * List out code formatting steps.
```

Comment:
Is this different than the "Coding Style" section? 
From the example, do you mean: "How to format your commit message" or "How to run your linter".

Reply:
No, not the commit message.  He didn't have a Coding Style section before, I added it.  So it might be duplicate.  I added some text to say that, as part of the PR process, just ensure the contributor has followed the Coding Style section.